packages/assets-controllers/src/TokenDetectionController.test.ts

```diff
@@ -122,6 +125,22 @@ const setupTokenListController = (
   return { tokenList, tokenListMessenger };
 };
 
+const buildTokenDetectionControllerMessenger: (
+  controllerMessenger: MainControllerMessenger,
+) => TokenDetectionControllerMessenger = (controllerMessenger) =>
+  controllerMessenger.getRestricted({
+    name: controllerName,
+    allowedActions: [
+      'NetworkController:getState',
+      'NetworkController:getNetworkClientById',
+      'TokenListController:getState',
+    ],
+    allowedEvents: [
+      'NetworkController:stateChange',
+      'TokenListController:stateChange',
+    ],
+  });
```

Comment:
Seems we could avoid the duplicate `(controllerMessenger) => ...` by making this a `function`:

``` suggestion
/**
 * Builds a messenger that TokenDetectionMessenger expects.
 *
 * @returns The messenger.
 */
function buildTokenDetectionControllerMessenger: (
  controllerMessenger: MainControllerMessenger,
): TokenDetectionControllerMessenger {
  return controllerMessenger.getRestricted({
    name: controllerName,
    allowedActions: [
      'NetworkController:getState',
      'NetworkController:getNetworkClientById',
      'TokenListController:getState',
    ],
    allowedEvents: [
      'NetworkController:stateChange',
      'TokenListController:stateChange',
    ],
  });
}

Reply:
Applied here: https://github.com/MetaMask/core/pull/3609/commits/7b3d4e45a3fef59e3bf9f0bbf1f6d7e749548000